app/models/blueprint.rb

```diff
@@ -26,27 +26,37 @@ def deep_copy(new_attributes = {})
   def create_bundle(options)
     options = options.with_indifferent_access
     self.class.transaction do
-      ServiceTemplate.create(
+      new_bundle = ServiceTemplate.create(
         :name         => name,
         :description  => description,
         :blueprint    => self,
         :service_type => 'composite'
-      ).tap do |new_bundle|
-        add_catalog_items(new_bundle, options[:service_templates]) if options.key?(:service_templates)
-        new_bundle.service_template_catalog = options[:service_catalog]
+      )
+      add_catalog_items(new_bundle, options[:service_templates]) if options.key?(:service_templates)
+      new_bundle.service_template_catalog = options[:service_catalog]
 
-        if options.key?(:service_dialog)
-          new_dialog = options[:service_dialog].deep_copy(:name => random_dialog_name(name)).tap(&:save!)
-        end
-        add_entry_points(new_bundle, options[:entry_points], new_dialog)
-
-        new_bundle.save!
+      if options.key?(:service_dialog)
+        new_dialog = options[:service_dialog].deep_copy(:label => random_dialog_name(name)).tap(&:save!)
       end
+      add_entry_points(new_bundle, options[:entry_points], new_dialog)
+
+      new_bundle.save!
+      reload.bundle
```

Comment:
It is discovered during spec tests. After the bundle is saved, `blueprint.bundle` still returns `nil` so I cannot continue to use the blueprint object as expected.